controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -172,7 +172,7 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 	}
 
 	// Override with MCG options specified in the storage cluster spec
-	if sc.Spec.MultiCloudGateway != nil {
+	if sc.Spec.MultiCloudGateway.DbStorageClassName != "" {
```

Comment:
you still need `nil` check as its a pointer

Reply:
done